src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -797,29 +804,32 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     // but using the iterator instead.
 
     const iteratorFn = getIteratorFn(newChildrenIterable);
-    if (typeof iteratorFn !== 'function') {
-      throw new Error('An object is not an iterable.');
-    }
+    invariant(
+      typeof iteratorFn === 'function',
+      'An object is not an iterable. (%s)',
```

Comment:
Unless user specified code returns something different the second time we read `@@iterator`, no?